src/test/file.rs

```diff
@@ -1,10 +1,680 @@
+                                }
+                            }
+                            "subtract" => {
+                                let modified_date = match m.unit.to_lowercase().as_str() {
+                                    "days" => dt.checked_sub_days(Days::new(mod_value)),
+                                    "weeks" => dt.checked_sub_days(Days::new(mod_value * 7)),
+                                    "months" => {
+                                        dt.checked_sub_months(Months::new(mod_value as u32))
+                                    }
+                                    // TODO: add support for years
+                                    _ => None,
+                                };
+
+                                if let Some(md) = modified_date {
+                                    dt = md;
+                                }
+                            }
+                            _ => {}
+                        }
+                    }
+                }
+                dt
+            })
+            .map(|d| self.time_to_str(&d))
+    }
+}
+
+impl Checker for DateSpecification {
+    type Item = String;
+    fn check(
+        &self,
+        val: &Self::Item,
+        formatter: &impl Fn(&str, &str) -> String,
+    ) -> Vec<Validated<(), String>> {
+        vec![
+            self.check_val(&val, formatter),
+            self.check_min(&val, formatter),
+            self.check_max(&val, formatter),
+            self.check_none_of(&val, formatter),
+            self.check_one_of(&val, formatter),
+        ]
+    }
+}
+
+#[derive(Hash, Serialize, Debug, Clone, Deserialize, PartialEq)]
+#[serde(rename_all = "camelCase")]
+#[serde(tag = "type")]
+pub enum DatumSchema {
+    Float {
+        #[serde(flatten)]
+        specification: Option<FloatSpecification>,
+    },
+    Int {
+        #[serde(flatten)]
+        specification: Option<Specification<i64>>,
+    },
+    String {
+        #[serde(flatten)]
+        specification: Option<Specification<String>>,
+    },
+    Date {
+        #[serde(flatten)]
+        date: DateSpecification,
+    },
+    List {
+        #[serde(skip_serializing_if = "Option::is_none")]
+        schema: Option<Box<DatumSchema>>,
+    },
+    Object {
+        #[serde(skip_serializing_if = "Option::is_none")]
+        schema: Option<BTreeMap<String, DatumSchema>>,
+    },
+}
+
+impl DatumSchema {
+    fn check(
+        &self,
+        actual: &serde_json::Value,
+        formatter: &impl Fn(&str, &str) -> String,
+    ) -> Vec<Validated<(), String>> {
+        trace!("validating datum");
+        let mut ret = self.check_float(actual, formatter);
+        ret.append(self.check_int(actual, formatter).as_mut());
+        ret.append(self.check_string(actual, formatter).as_mut());
+        ret.append(self.check_list(actual, formatter).as_mut());
+        ret.append(self.check_object(actual, formatter).as_mut());
+        ret
+    }
+    /*
+    fn check_date(
```

Comment:
is the date no longer expected to be here or has it just not been implemented yet?